lib/internal/streams/readable.js

```diff
@@ -290,11 +323,11 @@ function readableAddChunk(stream, chunk, encoding, addToFront) {
   if (err) {
     errorOrDestroy(stream, err);
   } else if (chunk === null) {
-    state.reading = false;
+    state.state &= ~kReading;
```

Comment:
if you keep this line and the rest of the assignment as is, wouldn't v8 just inline the setter?

I'm asking as it's not straightforward

Reply:
When you do the changes for Writable we'll go over the system analyzer together and see I basically did the changes based on generated IR/maps